app/models/volume_mapping.rb

```diff
@@ -17,6 +17,19 @@ class VolumeMapping < ApplicationRecord
   supports_not :delete
   acts_as_miq_taggable
 
+  def mapped_to
+    raise NotImplementedError
+  end
+
+  def name
+    mapped_type = mapped_to.class.base_class
+    if mapped_type == ApplicationRecord
```

Comment:
I was expecting this to more like `case mapped_type; when HostInitiator; when HostInitiatorGroup`
Is ApplicationRecord actually returned by base_class here?

Reply:
I'm not trying to differentiate between HostInitiator and HostInitiator Group.
I'm trying to get a pretty display name of the type. 
To get the pretty display name, I want to run `.display_name` on the baes class from `manageiq`. Not on the subclass from my provider. That's why I do `.base_class`.

The reason I then check that it's not `AppicationRecord` is for future-proofing. If anyone in the future adds a `mapped_to` object that works without a subtype, I don't want the algorithm to return the words `Application Record`.
This is also the reason why I don't explicitly check if it's HostInitiator or HostInitiatorGroup. I want it to be future-proof. When new types of `mapped_to` objects are added I don't want this code to need to change.

Reply:
Okay it might be better to push `name` down to the subclasses then, still not sure when you'd get a mapped type exactly `== ApplicationRecord` here but we can discuss in a follow-up